backend/src/main/java/com/redhat/cloud/notifications/routers/internal/InternalPermissionService.java

```diff
@@ -78,8 +80,21 @@ public InternalUserPermissions getPermissions() {
     @GET
     @Path("/")
     @Produces(MediaType.APPLICATION_JSON)
-    public List<InternalRoleAccess> getAccessList() {
-        return internalRoleAccessResources.getAll();
+    public List<InternalApplicationUserPermission> getAccessList() {
+        List<InternalRoleAccess> accessList = internalRoleAccessResources.getAll();
+        Set<UUID> applicationIds = accessList.stream().map(InternalRoleAccess::getApplicationId).collect(Collectors.toSet());
+        Map<UUID, String> applications = applicationResources
+                .getApplications(applicationIds)
+                .stream()
+                .collect(Collectors.toMap(Application::getId, Application::getDisplayName));
```

Comment:
This could be simpler if `InternalRoleAccess` contained a `@ManyToOne Application` field.